src/pages/workspace/categories/withPolicyAccessOrNotFound.tsx

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable rulesdir/no-negated-variables */
+import React, {useEffect} from 'react';
+import type {ForwardedRef, RefAttributes} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import FullscreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import getComponentDisplayName from '@libs/getComponentDisplayName';
+import Navigation from '@libs/Navigation/Navigation';
+import * as PolicyUtils from '@libs/PolicyUtils';
+import NotFoundPage from '@pages/ErrorPage/NotFoundPage';
+import * as Policy from '@userActions/Policy';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type * as OnyxTypes from '@src/types/onyx';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+
+type WithWorkspaceAccessOnyxProps = {
+    /** The report currently being looked at */
+    policy: OnyxEntry<OnyxTypes.Policy>;
+
+    /** Indicated whether the report data is loading */
+    isLoadingReportData: OnyxEntry<boolean>;
+};
+
+type WithWorkspaceAccessProps = WithWorkspaceAccessOnyxProps & {
+    /** The report currently being looked at */
+    route: {params: {policyID: string}};
+
+    /** The report currently being looked at */
+    policy: OnyxTypes.Policy;
+};
+
+export default function (): <TProps extends WithWorkspaceAccessProps, TRef>(
```

Comment:
AFAIK, we prefer hooks to hocs. So, can this logic be implemented via a hook?

Reply:
I think replacing this specific hoc to hook might not be the best option here, it manipulates the display screen using not found or fullscreen loading indicator

Reply:
Ah, got it. It makes sense then. 👍